pkg/task/display.go

```diff
@@ -5,71 +5,73 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"sort"
 
+	SemVer "github.com/Masterminds/semver"
+	"github.com/fatih/color"
 	"github.com/olekukonko/tablewriter"
 	"github.com/spf13/cobra"
 	"github.com/turbot/go-kit/files"
+	"github.com/turbot/steampipe/pkg/constants"
 	"github.com/turbot/steampipe/pkg/error_helpers"
 	"github.com/turbot/steampipe/pkg/filepaths"
+	"github.com/turbot/steampipe/pkg/plugin"
 	"github.com/turbot/steampipe/pkg/utils"
 )
 
 const (
-	NotificationsStructVersion = 20221121
+	AvailableVersionsCacheStructVersion = 20221121
 )
 
-type Notifications struct {
-	StructVersion      uint32   `json:"struct_version"`
-	CliNotifications   []string `json:"cli_notifications"`
-	PluginNotification []string `json:"plugin_notifications"`
+type AvailableVersionCache struct {
+	StructVersion uint32                               `json:"struct_version"`
+	CliCache      *CLIVersionCheckResponse             `json:"cli_version"`
+	PluginCache   map[string]plugin.VersionCheckReport `json:"plugin_version"`
 }
 
-func (n *Notifications) GetAll() []string {
-	return append(n.CliNotifications, n.PluginNotification...)
-}
-
-func (r *Runner) saveNotifications(cliNotificationsLines, pluginNotificationLines []string) error {
+func (r *Runner) saveAvailableVersions(cli *CLIVersionCheckResponse, plugin map[string]plugin.VersionCheckReport) error {
 	utils.LogTime("Runner.saveNotifications start")
 	defer utils.LogTime("Runner.saveNotifications end")
 
-	if len(cliNotificationsLines)+len(pluginNotificationLines) == 0 {
+	if cli == nil && len(plugin) == 0 {
 		// nothing to save
 		return nil
 	}
 
-	notifs := &Notifications{
-		StructVersion:      NotificationsStructVersion,
-		CliNotifications:   cliNotificationsLines,
-		PluginNotification: pluginNotificationLines,
+	notifs := &AvailableVersionCache{
+		StructVersion: AvailableVersionsCacheStructVersion,
+		CliCache:      cli,
+		PluginCache:   plugin,
 	}
 	// create the file - if it exists, it will be truncated by os.Create
-	f, err := os.Create(filepaths.NotificationsFilePath())
+	f, err := os.Create(filepaths.AvailableVersionsFilePath())
 	if err != nil {
 		return err
 	}
+	defer f.Close()
 	encoder := json.NewEncoder(f)
 	return encoder.Encode(notifs)
 }
 
-func (r *Runner) hasNotifications() bool {
+func (r *Runner) hasAvailableVersion() bool {
 	utils.LogTime("Runner.hasNotifications start")
 	defer utils.LogTime("Runner.hasNotifications end")
-	return files.FileExists(filepaths.NotificationsFilePath())
+	return files.FileExists(filepaths.AvailableVersionsFilePath())
 }
 
-func (r *Runner) getNotifications() (*Notifications, error) {
+func (r *Runner) getAvailableVersions() (*AvailableVersionCache, error) {
```

Comment:
is this function loading from cache only - which case maybe call it `loadCachedVersions`